src/mappings/events/vaults.ts

```diff
@@ -53,8 +53,84 @@ export const vaultsEventKit = (block: any, data: any, module: string, path: stri
     return promises;
   }
 
+  async function updateDailyVaultState(
+    oldState: string | undefined,
+    newState: string,
+    blockTime: Date,
+    blockHeight: number,
+  ): Promise<VaultStatesDaily> {
+    const dateKey = dateToDayKey(blockTime).toString();
+    let vaultState: VaultStatesDaily | undefined = await VaultStatesDaily.get(dateKey);
+
+    if (!vaultState) {
+      const yesterdayDate = new Date(blockTime);
+      yesterdayDate.setDate(yesterdayDate.getDate() - 1);
+      const yesterdayDateKey = dateToDayKey(yesterdayDate).toString();
+      const yesterdayVaultState: VaultStatesDaily | undefined = await VaultStatesDaily.get(yesterdayDateKey);
+
+      if (yesterdayVaultState) {
+        vaultState = new VaultStatesDaily(
+          dateKey,
+          BigInt(blockHeight),
+          blockTime,
+          yesterdayVaultState.active,
+          yesterdayVaultState.closed,
+          yesterdayVaultState.liquidating,
+          yesterdayVaultState.liquidated,
+          yesterdayVaultState.liquidatedClosed,
+        );
+      } else {
+        vaultState = new VaultStatesDaily(
+          dateKey,
+          BigInt(blockHeight),
+          blockTime,
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+        );
+      }
+    }
+
+    const propertyMap = {
+      [VAULT_STATES.ACTIVE]: 'active',
+      [VAULT_STATES.LIQUIDATED]: 'liquidated',
+      [VAULT_STATES.LIQUIDATING]: 'liquidating',
+      [VAULT_STATES.CLOSED]: 'closed',
+    };
+
+    const closedPropertyMap = {
+      [VAULT_STATES.ACTIVE]: 'closed',
+      [VAULT_STATES.LIQUIDATED]: 'liquidatedClosed',
+    };
+
```

Comment:
Do you think we should add `propertyMap` and `closedPropertyMap` in `constants.js`?

Reply:
its very specific to this function and i dont think it will be reused anywhere